File: we_5_6_gyrocompass_heading_accuracy.m
% Integrated Navigation Systems — Chapter 5: Inertial Navigation Mechanization
% Worked Example 5.6 — Gyrocompass heading accuracy
% Companion code (educational reference implementation; see the book for context).

OmH=15.041*cos(deg2rad(40)); dpsi=0.01/OmH;
fprintf('OmH=%.2f deg/hr, heading unc=%.3f deg\n',OmH,rad2deg(dpsi)); % 11.52, 0.050

File: we_5_6_gyrocompass_heading_accuracy.py
# Integrated Navigation Systems — Chapter 5: Inertial Navigation Mechanization
# Worked Example 5.6 — Gyrocompass heading accuracy
# Companion code (educational reference implementation; see the book for context).

import numpy as np
OmH=15.041*np.cos(np.radians(40)); dpsi=0.01/OmH
print(round(OmH,2),'deg/hr', round(np.degrees(dpsi),3),'deg')  # 11.52 0.050
